### PR TITLE
Various form layout updates (incl. vertical layout)

### LINKED ISSUE #3529: Retire horizontal form layout and use vertical layout only
#### Summary
As part of the form UX improvements (#2813), we will move towards vertical form layout for all our forms (meaning to display the field label above the field).

#### Why do we need this?
There are a couple of reasons to make this shift:
- Horizontal layout is harder to scan, since there's a high distance between the field label and field input
- Horizontal layout puts a strain on users for having to remember the field title (and purpose) while making an input
- Horizontal layout has some layout inconsistencies/issues for checkboxes, due to the fact that checkboxes usually don't need a field title (they just use the label next to the checkbox)
- Horizontal layout is not as common as vertical layout
- Horizontal layout puts a width limit on labels
- There's also a technical benefit of not having to deal with two different layout modes for fields (as we do now)

#### What is already there? What do you see now?
Luckily, our forms already allow for vertical layouts (like we currently use in the login form). Right now the horizontal layout is the default.

#### What is missing? What do you want to see?
- Removing the `horizontal` prop flag in our `<Form />` component
- Removing additional styling that corresponds to the horizontal layout
- Some alternative layout solutions for
  - Displaying checkboxes without field title (and use the checkbox label instead)
  - Same for `<Input.Toggled />`
  - Still enabling the use of tooltip icons for these fields too
- Some styling adjustments
  - Form sub titles need to be more distinct from field labels 
  - Overall spacing adjustments
  - Allow for different field widths (e.g. for fields such as "Port number")

I've looked through all of our forms and don't see any big issues that require broader restructure to make this change to the vertical form layout.

#### How do you propose to implement this?
This is a matter of some styling and display logic changes. I think the most challenging here is to enable some different stylings in the `<Form.Field />` component without stirring it up too much. All in all, this should not be a big issue though.

#### How do you propose to test this?
These changes should not affect our existing tests, since they'll mostly use semantic selectors, which we won't change. I don't think this change needs any new tests introduced.

#### Can you do this yourself and submit a Pull Request?
Yes.

### PR DESCRIPTION
**Pull request by kschiffer**
#### Summary
Closes #3529 
References #2813 

image

#### Changes
- Remove horizontal layout
  - Remove horizontal props from forms and inputs (except checkbox/radio button groups)
  - Remove styling for horizontal forms
- Replace literal headings with `<Form.SubTitle />`
- Allow fields without title (to be used e.g. by checkboxes)
- Allow extending checkboxes with children prop (to add glossary icon)
- Introduce `inputWidth` prop for some inputs (to easily set various input widths for fields)
- Various related styling fixes and tweaks
- Remove `tabindex` for glossary icons
- Some low-hanging-fruit fixes to some forms
  - Updated messages
  - Updated form structure and field order
- Updated stories

#### Testing

Manual testing as well as existing e2e tests.

##### Regressions

This should not introduce any regressions, since this is mostly styling focused

#### Notes for Reviewers
See #3529 and #2813 for details

#### Checklist

- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
- [ ] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [ ] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/webui/components/checkbox/checkbox.js [hunk 1]
```diff
@@ -171,6 +174,7 @@ class Checkbox extends React.PureComponent {
           <span className={style.checkmark} />
         </span>
         {label && <Message className={style.label} content={label} />}
+        {children}
```
Comment (bafonins) [inline pkg/webui/components/checkbox/checkbox.js]: Any reason for adding this? Cant find any usage of `<Checkbox />` with `children` being set.

File: pkg/webui/components/link/index.js [hunk 2]
```diff
@@ -135,6 +135,7 @@ const DocLink = function(props) {
     secondary,
     primary,
     disabled,
+    tabIndex,
```
Comment (bafonins) [inline pkg/webui/components/link/index.js:142]: `tabIndex` is not defined in prop-types

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,8 @@ For details about compatibility between different releases, see the **Commitment
 ### Added
 
 ### Changed
+- Form field layouts in the Console (field labels are now dispayed above the field).
```
Comment (bafonins) [inline CHANGELOG.md:18]: new line missing

**Review by bafonins — COMMENTED** [review]
Looks good 👌

Is the difference between field and submit bar lengths intentional? This is how the application form looks on my screen:
Screenshot 2020-12-08 at 17 31 24

File: pkg/webui/components/checkbox/checkbox.js (hunk 1 above)
Comment (kschiffer) [reply pkg/webui/components/checkbox/checkbox.js]: Ah yeah, we'll use this later on to be able to display tooltip icons after a checkbox label. If you want I can remove it here and re-add it later, when we use it.

Comment (bafonins) [reply pkg/webui/components/checkbox/checkbox.js]: I would prefer adding this together with tooltips

**Review by bafonins — APPROVED** [review]

**Pull request merged**